inc/inventorylog.class.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+/**
+ * -------------------------------------------------------------------------
+ * SCCM plugin for GLPI
+ * -------------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of SCCM.
+ *
+ * SCCM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * SCCM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with SCCM. If not, see <http://www.gnu.org/licenses/>.
+ * -------------------------------------------------------------------------
+ * @author    François Legastelois
+ * @copyright Copyright (C) 2014-2023 by SCCM plugin team.
+ * @license   GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
+ * @link      https://github.com/pluginsGLPI/sccm
+ * -------------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+    die("Sorry. You can't access directly to this file");
+}
+
+use Glpi\Application\View\TemplateRenderer;
+
+class PluginSccmInventoryLog extends \CommonDBTM
+{
+
+    public $dohistory = true;
+    public static $rightname = 'config';
+
+    public const SCCM_STATE_DONE = "sccm-done";
+    public const SCCM_STATE_FAIL = "sccm-fail";
+
+    static function install(Migration $migration)
+    {
+        global $DB;
+
+        $table = PluginSccmInventoryLog::getTable();
+
+        if (!$DB->tableExists($table)) {
+            $migration->displayMessage("Installing $table");
+            $query = "CREATE TABLE $table (
+                `id` int(11) NOT NULL AUTO_INCREMENT,
```

Comment:
Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_sccm_inventorylogs`.`id`